mesheryctl/cmd/perf.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2019 The Meshery Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	testURL  = ""
+	testName = ""
+	testMesh = ""
+	// testFile         = ""
+	qps              = ""
+	parallelRequests = ""
+	testDuration     = ""
+	loadGenerator    = ""
+	testCookie       = ""
+)
+
+var seededRand *rand.Rand = rand.New(
+	rand.NewSource(time.Now().UnixNano()))
+
+// Generates random string
+func StringWithCharset(length int) string {
+	const charset = "abcdefghijklmnopqrstuvwxyz" + "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	b := make([]byte, length)
+	for i := range b {
+		b[i] = charset[seededRand.Intn(len(charset))]
+	}
+	return string(b)
+}
```

Comment:
Will the /load-test-smps API generate a random name for the test if one is not provided?

Reply:
I shall look into it.

Reply:
No actually, the server responds by saying `Provide a name for the test`.

Reply:
I saw the same behavior in my tests earlier today, too. Interesting as I suppose Meshery's frontend generates a unique value when tests are invoked in the UI. That random name generator functionality should ultimately sit server-side with Meshery.

Reply:
So shall I add a random name generator in the server side?

Reply:
@kanishkarj Why is there a random name generator? AFAIK, in the UI we can enter the test name, won't this create confusion when the user sees the test results? Or am I getting something over here ?

Reply:
oh yes, we wanted to handle the case where the user doesn't provide a test name. let's say the user doesn't provide a name while calling the CLI, then we call the random string generator to get a random name, and use that.

Actually, we still aren't sure if this random generation should happen server side or client-side. I put it here for now.